src/types/enums.rs

```diff
@@ -104,6 +104,66 @@ enum_type_set! {
 }
 
 impl<'ctx> BasicMetadataTypeEnum<'ctx> {
+    /// Creates a [`FunctionType`] with this [`BasicMetadataTypeEnum`] for its return type.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use inkwell::context::Context;
+    /// use inkwell::types::BasicMetadataTypeEnum;
+    ///
+    /// let context = Context::create();
+    /// let ty: BasicMetadataTypeEnum = context.i8_type().into();
+    /// let fn_type = ty.fn_type(&[], false);
+    /// ```
+    #[llvm_versions(6.0..=latest)]
+    pub fn fn_type(self, param_types: &[BasicMetadataTypeEnum<'ctx>], is_var_args: bool) -> FunctionType<'ctx> {
+        use BasicMetadataTypeEnum::*;
+        match self {
+            ArrayType(ty) => ty.fn_type(param_types, is_var_args),
+            FloatType(ty) => ty.fn_type(param_types, is_var_args),
+            IntType(ty) => ty.fn_type(param_types, is_var_args),
+            PointerType(ty) => ty.fn_type(param_types, is_var_args),
+            StructType(ty) => ty.fn_type(param_types, is_var_args),
+            VectorType(ty) => ty.fn_type(param_types, is_var_args),
+            // TODO: are functions really able to return metadata?
```

Comment:
We should verify this TODO before adding this function. If it's not valid, it should return a Result

Reply:
Ok, I'll check it

Reply:
I tried verification of module with function has metadata as return type. It works. But I never found anything about returning metadata on official documentation.

Reply:
Any ideas what to do?

Reply:
If it passed verification then I suppose that's fine